kernel/src/table_changes/mod.rs

```diff
@@ -176,6 +186,63 @@ impl TableChanges {
     }
 }
 
+/// Ensures that change data feed is enabled in `table_properties`.
+///
+/// Performing change data feed on  tables with column mapping is currently disallowed.
+/// This will be less restrictive in the future. Because column mapping is disallowed, we also
+/// check that column mapping is disabled, or the column mapping mode is `None`.
+fn check_cdf_table_properties(table_properties: &TableProperties) -> DeltaResult<()> {
+    require!(
+        table_properties.enable_change_data_feed.unwrap_or(false),
+        Error::unsupported("Change data feed is not enabled")
+    );
+    require!(
+        matches!(
+            table_properties.column_mapping_mode,
+            None | Some(ColumnMappingMode::None)
+        ),
+        Error::unsupported("Change data feed not supported when column mapping is enabled")
+    );
+    Ok(())
+}
+
+/// Ensures that Change Data Feed is supported for a table with this [`Protocol`] .
+//
+//  Currently the only read feature allowed is deletion vectors. This will be expanded in the
+//  future to support more delta table features.
+//
+//  Because only deletion vectors are supported, reader version 2 will not be allowed. That is
+//  because version 2 requires that column mapping is enabled. Reader versions 1 and 3 are allowed.
```

Comment:
Currently, we just choose to only support the narrowest set of features (ie DVs). That PR won't fix this issue since column mapping support for CDF also requires:
- Tests that exercises CDF on tables with column mapping enabled
- Logic to handle any changes to column mapping mode

I'll put up an issue to address the CDF specific concerns.

Reply:
https://github.com/delta-io/delta-kernel-rs/issues/560